cips/cip-X.md

```diff
@@ -0,0 +1,271 @@
+}
+```
+
+Signed MsgSignIn json example:
+
+```json
+{
+  "type": "cosmos-sdk/StdTx",
+  "value": {
+    "msg": [
+      {
+        "type": "offchain/MsgSignIn",
+        "value": {
+          "appDomain": "exampleSwap",
+          "uri": "https://exampleSwap.com/login",
+          "signerAddress": "cosmos1hftz5ugqmpg9243xeegsqqav62f8hnywsjr4xr",
+          "nonce": "14368412",
+          "issued-at": "2023-10-10T16:25:24Z"
+        }
+      }
+    ],
+    "fee": {
+      "amount": [],
+      "gas": "0"
+    },
+    "signatures": [
+      {
+        "pub_key": {
+          "type": "tendermint/PubKeySecp256k1",
+          "value": "AqnDSiRoFmTPfq97xxEb2VkQ/Hm28cPsqsZm9jEVsYK9"
+        },
+        "signature": "8y8i34qJakkjse9pOD2De+dnlc4KvFgh0wQpes4eydN66D9kv7cmCEouRrkka9tlW9cAkIL52ErB+6ye7X5aEg=="
+      }
+    ],
+    "memo": ""
+  }
+}
+```
+
+## Drawbacks
+
+This CIP requires a fixed relationship between an account address and a public key. That means it won't work if [ADR-034](https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-rc.1/docs/architecture/adr-034-account-rekeying.md) 
+is implemented.
+
+Doesn't work with multisig accounts.
+
+## Rationale
+
+By creating different messages for different use cases, this design fulfills the current demands of the community for
+[arbitrary data signing](https://github.com/cosmos/cosmos-sdk/issues/4581) and [human-readable sign-in message](https://github.com/cosmos/cosmos-sdk/pull/7896#issuecomment-1125254734).
+
+Implementing this design aligns with the long-standing requests from the community. Several zones have implemented their
+own arbitrary messages based on ADR-036:
+
+ + [Sommelier](https://github.com/cosmos/cosmos-sdk/pull/7727#issuecomment-821201911)
+ + [Aleph.im](https://github.com/cosmos/cosmos-sdk/pull/7727#issuecomment-821776061)
+ + [Secret](https://github.com/scrtlabs/cosmos-sdk/compare/v0.9.1-scrt...v0.9.2-scrt)
+
+Even some work have been done on [cosmjs](https://github.com/cosmos/cosmjs/pull/847).
+
+There are zones and applications that prove account ownership by signing a random challenge with ADR-036 specification.
+
+## Prior Art
+
+Other blockchains such as Ethereum with [eip-4361](https://eips.ethereum.org/EIPS/eip-4361) or Solana with its
+[Off-chain message signing](https://docs.solana.com/proposals/off-chain-message-signing) offer standard methods for
+application sign-in.
+
+There was a prior attempt to implement this in the sdk with [ADR-036](https://github.com/cosmos/cosmos-sdk/blob/v0.50.0-rc.1/docs/architecture/adr-036-arbitrary-signature.md).
+However, a consensus was not achieved on ADR-036, as the community demanded a way for sign-in method that ADR-036
+did not provide. This is the main reason for defining two different messages in this CIP. One for arbitrary data and
+the other for sing-in.
+
+
+## Unresolved Questions
+
+ + Is defining different messages per use case the best approach, or could a protobuf with a `oneof` be sufficient? 
```

Comment:
i dont think either need to happen. The message can be encoded and added to the msgarbitray signing right?

Reply:
for this usecase i think thats fine. We dont want to overcomplicate the system. The user will know whats being signed in the browser and would be able to verify the humanreadable bytes match the client